File: frontend/types/incidentes.ts
export interface IncidenteActivoVinculo {
  uuid: string;
  id_incidente: string;
  id_activo: string;
  notas: string | null;
  tipo_activo_registrado: string | null;
  sede_registrada: string | null;
  hostname_registrado: string | null;
  created_at: string | null;
  updated_at: string | null;
  hostname_actual: string | null;
  activo_eliminado: boolean | null;
}

export interface Incidente {
  uuid: string;
  titulo: string;
  descripcion: string | null;
  id_tipo: number;
  id_prioridad: number;
  id_estado: number;
  id_usuario_asignado: string;
  id_sede: number;
  fecha_cierre: string | null;
  eliminado: boolean;
  created_at: string | null;
  updated_at: string | null;
  tipo_nombre: string | null;
  prioridad_nivel: string | null;
  estado_nombre: string | null;
  activos_vinculados?: IncidenteActivoVinculo[];
}

export interface IncidenteCreate {
  titulo: string;
  descripcion?: string | null;
  id_tipo: number;
  id_prioridad: number;
  id_estado: number;
  id_usuario_asignado?: string | null;
  activos?: string[];
}

export interface IncidenteUpdate {
  titulo?: string;
  descripcion?: string | null;
  id_tipo?: number;
  id_prioridad?: number;
  id_estado?: number;
  id_usuario_asignado?: string | null;
  fecha_cierre?: string | null;
  eliminado?: boolean;
}

export interface IncidenteActivoLinkRequest {
  id_activo: string;
  notas?: string | null;
}

File: backend/app/schemas/incidentes.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.incidente_activos import IncidenteActivoOut


class IncidenteCreate(BaseModel):
    titulo: str = Field(min_length=3, max_length=255)
    descripcion: str | None = None
    id_tipo: int
    id_prioridad: int
    id_estado: int
    id_usuario_asignado: UUID | None = None
    activos: list[UUID] = Field(default_factory=list)


class IncidenteUpdate(BaseModel):
    titulo: str | None = Field(default=None, min_length=3, max_length=255)
    descripcion: str | None = None
    id_tipo: int | None = None
    id_prioridad: int | None = None
    id_estado: int | None = None
    id_usuario_asignado: UUID | None = None
    fecha_cierre: datetime | None = None
    eliminado: bool | None = None


class IncidenteOut(BaseModel):
    uuid: UUID
    titulo: str
    descripcion: str | None
    id_tipo: int
    id_prioridad: int
    id_estado: int
    id_usuario_asignado: UUID
    id_sede: int
    fecha_cierre: datetime | None
    eliminado: bool
    created_at: datetime | None = None
    updated_at: datetime | None = None
    tipo_nombre: str | None = None
    prioridad_nivel: str | None = None
    estado_nombre: str | None = None
    activos_vinculados: list[IncidenteActivoOut] = Field(default_factory=list)
